stability: segfault in raft/rawnode

`blue` crashed on segfault in raft:
sha: 07705f2
node: `40.123.43.229`

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10389f3]

goroutine 98 [running]:
panic(0x1759dc0, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420388120)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x6e
panic(0x1759dc0, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*RawNode).ReportUnreachable(0x0, 0x3)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/rawnode.go:243 +0x83
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).tickRaftMuLocked(0xc434197800, 0x1aa9100, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2533 +0x14f
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).tick(0xc434197800, 0xc420471b00, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2519 +0x85
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processTick(0xc420898700, 0x1bd2, 0xc439eadf00)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3268 +0xfb
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42095a000, 0xc420388120)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:225 +0x2b6
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420388120, 0xc420e7cd60)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66
```

This looks like @petermattis's build, but this comes after the vendoring change, apparently with an update to raft.

--- a/pkg/storage/replica.go
+++ b/pkg/storage/replica.go
@@ -2536,6 +2536,12 @@ func (r *Replica) tickRaftMuLocked() (bool, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	// If the raft group is uninitialized, do not initialize raft groups on
+	// tick.
+	if r.mu.internalRaftGroup == nil {
+		return false, nil
+	}
+
 	r.unreachablesMu.Lock()
 	remotes := r.unreachablesMu.remotes
 	r.unreachablesMu.remotes = nil
@@ -2545,11 +2551,6 @@ func (r *Replica) tickRaftMuLocked() (bool, error) {
 		r.mu.internalRaftGroup.ReportUnreachable(uint64(remoteReplica))
 	}
 
-	// If the raft group is uninitialized, do not initialize raft groups on
-	// tick.
-	if r.mu.internalRaftGroup == nil {
-		return false, nil
-	}
 	if r.mu.quiescent {
 		// While a replica is quiesced we still advance its logical clock. This is
 		// necessary to avoid a scenario where the leader quiesces and a follower

